spinalcordtoolbox/registration/algorithms.py

```diff
@@ -447,56 +448,22 @@ def register_dl_multimodal_cascaded_reg(fname_src, fname_dest, fname_warp_forwar
         nb_unet_features=([256, 256, 256, 256], [256, 256, 256, 256, 256, 256])
     )
     logger.info("\n Creating VxmDense models with arguments:")
-    logger.info(f"\n{reg_args}")
+    logger.info(f"{reg_args}")
 
-    # ---- First Model ---- #
-    logger.info("\n Loading first VxmDense model...")
-    pt_first_model = vxm.networks.VxmDense(**reg_args)
-    model1_path = sct_dir_local_path('data', 'deepreg_models', 'pt_cascaded_first_model.pt')
-    trained_state_dict_first_model = torch.load(model1_path)
-    # Load the weights to the PyTorch model
-    weights_first_model = []
-    for k in trained_state_dict_first_model:
-        weights_first_model.append(trained_state_dict_first_model[k])
-    i = 0
-    i_max = len(list(pt_first_model.named_parameters()))
-    torchparam = pt_first_model.state_dict()
-    for k, v in torchparam.items():
-        if i < i_max:
-            torchparam[k] = weights_first_model[i]
-            i += 1
-    pt_first_model.load_state_dict(torchparam)
-    pt_first_model.to(device)
-    pt_first_model.eval()
-
-    # ---- Second Model ---- #
-    logger.info("\n Loading second VxmDense model...")
-    pt_second_model = vxm.networks.VxmDense(**reg_args)
-    model2_path = sct_dir_local_path('data', 'deepreg_models', 'pt_cascaded_second_model.pt')
-    trained_state_dict_second_model = torch.load(model2_path)
-    # Load the weights to the PyTorch model
-    weights_second_model = []
-    for k in trained_state_dict_second_model:
-        weights_second_model.append(trained_state_dict_second_model[k])
-    i = 0
-    i_max = len(list(pt_second_model.named_parameters()))
-    torchparam = pt_second_model.state_dict()
-    for k, v in torchparam.items():
-        if i < i_max:
-            torchparam[k] = weights_second_model[i]
-            i += 1
-    pt_second_model.load_state_dict(torchparam)
-    pt_second_model.to(device)
-    pt_second_model.eval()
+    volume_m = (new_img_shape[0] * new_img_shape[1] * new_img_shape[2] / 1e6)
+    ram_estimated = round((volume_m * 3.52) + 0.0538, 2)
+    ram_free = round(psutil.virtual_memory().free / 1024 ** 3, 2)
+    if ram_estimated + 1 > ram_free:
+        logger.warning(f"\nWARNING: System has {ram_free}GB of free memory, but inference may require as much as "
+                       f"{ram_estimated}GB to register images of size {new_img_shape}. As a result, the script may "
+                       f"quit unexpectedly due to lack of memory. To reduce memory requirements, please consider "
+                       f"cropping your images around the spinal cord prior to registration.")
```

Comment:
I've added this warning that tries to estimate when RAM issues will occur:

![image](https://user-images.githubusercontent.com/16181459/203118349-bac51dc6-47f7-40a3-826c-d86948e9e96e.png)

The estimated RAM value is based off of measurements taken by varying the size of the input image (across each dimension), then using `psutil` to check free RAM before and after inference.

Reply:
Thankfully, RAM appears to vary linearly with total volume, regardless of which dimension is varied:

![image](https://user-images.githubusercontent.com/16181459/203119238-93e7293b-6273-4e1d-9454-9d07b7d3bdca.png)

Hence the equation used above:

```python
    volume_m = (new_img_shape[0] * new_img_shape[1] * new_img_shape[2] / 1e6)
    ram_estimated = round((volume_m * 3.52) + 0.0538, 2)
```

Reply:
> ```python
> if ram_estimated + 1 > ram_free:
> ```

Note that I've made the check a little more sensitive than it actually is (by `~1GB`) in an attempt to warn users when they get close to the limit, even if they don't exceed it.

Reply:
awesome! very elegant solution, we are all Math lovers here